comictaggerlib/issueidentifier.py

```diff
@@ -543,142 +493,188 @@ def search(self) -> list[IssueResult]:
 
-        best_score: int = self.match_list[0].distance
-
-        if best_score >= self.min_score_thresh:
+        cover_matching_2 = []
+        final_cover_matching = cover_matching_1
+        if cover_matching_1[0].distance >= self.min_score_thresh:
             # we have 1 or more low-confidence matches (all bad cover scores)
             # look at a few more pages in the archive, and also alternate covers online
             self.log_msg("Very weak scores for the cover. Analyzing alternate pages and covers...")
-            hash_list = [cover_hash]
-            if narrow_cover_hash is not None:
-                hash_list.append(narrow_cover_hash)
-            for page_index in range(1, min(3, ca.get_number_of_pages())):
-                image_data = ca.get_page(page_index)
-                page_hash = self.calculate_hash(image_data)
-                hash_list.append(page_hash)
-
-            second_match_list = []
-            counter = 2 * len(self.match_list)
-            for m in self.match_list:
-                if self.progress_callback is not None:
-                    self.progress_callback(counter, len(self.match_list) * 3)
-                    counter += 1
-                self.log_msg(f"Examining alternate covers for ID: {m.series_id} {m.series}:")
-                try:
-                    score_item = self.get_issue_cover_match_score(
-                        m.image_url,
-                        m.alt_image_urls,
-                        hash_list,
-                        use_remote_alternates=True,
-                    )
-                except Exception:
-                    logger.exception("failed examining alt covers")
-                    self.match_list = []
-                    return self.match_list
-                self.log_msg(f"--->{score_item['score']}")
-                self.log_msg("")
-
-                if score_item["score"] < self.min_alternate_score_thresh:
-                    second_match_list.append(m)
-                    m.distance = score_item["score"]
-
-            if len(second_match_list) == 0:
-                if len(self.match_list) == 1:
+
+            temp = self._match_covers(terms, images + extra_images, issues, use_alternates=True)
+            for score in temp:
+                if score.distance < self.min_alternate_score_thresh:
+                    cover_matching_2.append(score)
+
+            if len(cover_matching_2) == 0:
+                if len(cover_matching_1) == 1:
                     self.log_msg("No matching pages in the issue.")
                     self.log_msg("--------------------------------------------------------------------------")
-                    print_match(self.match_list[0])
+                    self._print_match(cover_matching_1[0])
                     self.log_msg("--------------------------------------------------------------------------")
                     self.search_result = self.result_found_match_but_bad_cover_score
```

Comment:
This returns to alter `self.search_result` but there are two problems. L:660 would overwrite it if it didn't use `search_result` instead. So a low confidence result becomes a good one.

Reply:
Fixed